### ISSUE TITLE
Local import errors not reported, cause environment overrides not to be loaded

### ISSUE THREAD

**Issue by zagy** [flyingcircusio/batou#129]
In a deployment I found something like 
```python
from components.utils.foo import Bar
```

This magically works on the remote side, but not locally. The error is being added to the environment's exceptions (https://github.com/flyingcircusio/batou/blob/master/src/batou/environment.py#L124) but that's never reported.

The environment overrides are being computed locally (due to secrets) and shipped over to the remote side. Interestingly no overrides for the affected component are shipped to the remote side from the *environment* but from the secrets they are – which is even more confusing. 

On the remote side importing works, and deploying works. So the only result you actually see is that the environment overrides for this one component do not work.

**Comment by ctheune** [comment]
Can you poke me about the specific scenario? I'm having a really hard time reproducing this ...

**Comment by zagy** [comment]
The problem is that someone added the import to the environment and it semi-worked. We actually have `batou.c.` for the components, which is not very pythonic. But that's not the actual issue. The import should either work or fail in all cases.

**Comment by sallner** [comment]
So I have looked at the repository in question and investigated the problem. Some findings. 

* The specific issue occurred on a batou 2.0.2 (would be helpful next time in the ticket description @zagy)
* The repository introduced some `__init__.py` files to make the folders to modules, which should not be done. But removing those, the problem still persists. 
* If the import `from component.utils.component import ImportDummy` is present. The clear-text environment did not get overridden. A deployment to a vagrant env succeeded. So it is not necessary to have a real deployment. 
* If the import was removed, everything was fine.

I tried to reproduce this steps on a batou 2.3b2 and did not get a successful run on vagrant. The error looks also completely different and occurs already at the beginning of the configuration.

```
ERROR: Failed loading component file
      File: /.../batou-demo/components/test/component.py
 Exception: No module named 'components'

```
I would therefore like to write some documentation on using batou.c and afterwards close this issue as fixed in version 2.3. Any objections?

**Comment by zagy** [comment]
@sallner yeah, sure. If the error is consistently reported now, that's fine.
